cmd/sovereignnode/chainSimulator/tests/bridge/incoming_test.go

```diff
@@ -0,0 +1,176 @@
+
+	receiverWallet, err := cs.GenerateAndMintWalletAddress(core.SovereignChainShardId, chainSim.ZeroValue)
+	require.Nil(t, err)
+
+	headerNonce := uint64(9999999)
+	prevHeader := createHeaderV2(headerNonce, generateRandomHash(), generateRandomHash())
+	var txsEvent []*transaction.Event
+
+	for i := 0; i < 3; i++ {
+		prevHeader = addNextHeader(t, cs, &headerNonce, prevHeader, txsEvent)
+
+		if i == 0 {
+			txsEvent = createTransactionsEvent(receiverWallet.Bytes)
+		} else {
+			txsEvent = nil
+		}
+	}
+
+	err = cs.GenerateBlocks(1)
+	require.Nil(t, err)
+
+	esdts, _, err := nodeHandler.GetFacadeHandler().GetAllESDTTokens(receiverWallet.Bech32, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotNil(t, esdts)
+	require.True(t, esdts[token] != nil)
+	require.Equal(t, amountToTransfer, esdts[token].Value.String())
+}
+
+func addNextHeader(t *testing.T, cs chainSim.ChainSimulator, headerNonce *uint64, prevHeader *block.HeaderV2, txsEvent []*transaction.Event) *block.HeaderV2 {
+	err := cs.GenerateBlocks(1)
+	require.Nil(t, err)
+
+	nodeHandler := cs.GetNodeHandler(core.SovereignChainShardId)
+	incomingHeader := createIncomingHeader(nodeHandler, headerNonce, prevHeader, txsEvent)
+	err = nodeHandler.GetIncomingHeaderSubscriber().AddHeader(generateRandomHash(), incomingHeader)
+	require.Nil(t, err)
+
+	return incomingHeader.Header
+}
+
+func createIncomingHeader(nodeHandler process.NodeHandler, headerNonce *uint64, prevHeader *block.HeaderV2, txsEvent []*transaction.Event) *sovereign.IncomingHeader {
+	*headerNonce++
+	prevHeaderHash, _ := core.CalculateHash(nodeHandler.GetCoreComponents().InternalMarshalizer(), nodeHandler.GetCoreComponents().Hasher(), prevHeader)
+	incomingHeader := &sovereign.IncomingHeader{
+		Header:         createHeaderV2(*headerNonce, prevHeaderHash, prevHeader.GetRandSeed()),
+		IncomingEvents: txsEvent,
+	}
+	return incomingHeader
+}
+
+func createHeaderV2(nonce uint64, prevHash []byte, prevRandSeed []byte) *block.HeaderV2 {
+	return &block.HeaderV2{
+		Header: &block.Header{
+			PrevHash:     prevHash,
+			Nonce:        nonce,
+			Round:        nonce,
+			RandSeed:     generateRandomHash(),
+			PrevRandSeed: prevRandSeed,
+			ChainID:      []byte(configs.ChainID),
+		},
+	}
+}
+
+func generateRandomHash() []byte {
+	randomBytes := make([]byte, hashSize)
+	_, _ = rand.Read(randomBytes)
+	return randomBytes
+}
+
+func createTransactionsEvent(receiver []byte) []*transaction.Event {
+	codec := createDataCodec()
+	tokenData, _ := codec.SerializeTokenData(createTokenData())
+	eventData, _ := codec.SerializeEventData(createEventData())
+
+	events := make([]*transaction.Event, 0)
+	return append(events, &transaction.Event{
+		Identifier: []byte(eventIDDepositIncomingTransfer),
+		Topics:     [][]byte{[]byte(topicIDDepositIncomingTransfer), receiver, []byte(token), []byte(""), tokenData},
+		Data:       eventData,
+	})
+}
+
+func createDataCodec() dataCodec.SovereignDataCodec {
```

Comment:
You should have this from sovereignRunType, not create it here and also not always create it for each func call.